src/vs/base/test/node/powershell.test.ts

```diff
@@ -0,0 +1,68 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import * as assert from 'assert';
+import * as platform from 'vs/base/common/platform';
+import * as fs from 'fs';
+import { enumeratePowerShellInstallations, getFirstAvailablePowerShellInstallation, IPowerShellExeDetails } from 'vs/base/node/powershell';
+
+function checkPath(exePath: string) {
+	// Check to see if the path exists
+	let pathCheckResult = false;
+	try {
+		const stat = fs.statSync(exePath);
+		pathCheckResult = stat.isFile();
+	} catch {
+		// fs.exists throws on Windows with SymbolicLinks so we
+		// also use lstat to try and see if the file exists.
+		try {
+			pathCheckResult = fs.statSync(fs.readlinkSync(exePath)).isFile();
+		} catch {
+
+		}
+	}
+
+	assert.strictEqual(pathCheckResult, true);
+}
+
+if (platform.isWindows) {
+	suite('PowerShell finder', () => {
+
+		test('Can find first available PowerShell', async () => {
+			const pwshExe = await getFirstAvailablePowerShellInstallation();
+			const exePath = pwshExe?.exePath;
+			assert.notStrictEqual(exePath, null);
+			assert.notStrictEqual(pwshExe?.displayName, null);
+
+			checkPath(exePath!);
+		});
+
+		// In Azure DevOps or GitHub Actions, they have 3 PowerShell's available
+		// on Windows:
+		// 1. PowerShell stable
+		// 2. PowerShell as a .NET Global Tool
+		// 2. Windows PowerShell (x64)
+		// 3. Windows PowerShell (x86)
+		// Only run this test in CI where the result is predictable.
+		if (process.env.TF_BUILD || process.env.CI) {
```

Comment:
Predictable until we change/update the image of the test machine. Maybe instead the test should target Windows 10+ and expect that all default shells shipped with Windows 10 exist to future proof it?

Reply:
Yeah I guess so. I've refactored so the tests runs local as well... but I kept a small check in that makes sure there is 3 or more in AzDO or Actions